frontendservice/src/views/ShopOwnerHome.vue

```diff
@@ -1,79 +1,170 @@
-    <p>
-      This page will be Home page for the shop-owners. 
-    </p>
-    <p>
-      ToDo: Implement ShopOwnerHome page features including NavBar, Search Bar, MyShops and AddShop.
-    </p> 
+  <div>
     <div v-if="isVerified">
-      <router-link to="/create-shop">Create new shop</router-link>
-    </div>
-    <div v-if="isVerified">
-      <router-link to="/my-shops">My Shops</router-link>
+      <shopOwnerNavbar />
+      <h1 id="heading" v-if="isVerified">My Shops</h1>
+      <input type="search" v-model="searchResult" placeholder="search shops" id="searchbar" />
+      <table class="table">
+        <thead>
+          <th>Shop Name</th>
+          <th>Shop Type</th>
+          <th>All Tokens of Shop</th>
+          <th>Delete</th>
+        </thead>
+        <tbody>
+          <tr v-for="shop in filteredShops" :key="shop['shop'].shopId">
+            <td>
+              <a v-bind:href="'/shop-owner/shops/'+ shop['shop'].shopId" :shopId="shop['shop'].shopId">{{shop['shop'].shopName}}</a>
+            </td>
+            <td>{{shop['shop'].shopType}}</td>
+            <td v-if="shop['shop'].status=='ACTIVE'">
+              <a v-bind:href="'/shop-owner/shops/'+ shop.shopId" :id="shop.shopId">{{shop.numberOfActiveTokens}}</a>
+            </td>
+            <td v-if="shop['shop'].status!='ACTIVE'">
+              <a v-bind:href="'/shop-owner/shops/'+ shop.shopId" :id="shop.shopId">0</a>
+            </td>
+            <td v-if="shop['shop'].status=='ACTIVE'">
+              <button @click="deleteshop(shop['shop'].shopId)">x</button>
+            </td>
+            <td v-if="shop['shop'].status!='ACTIVE'">
+              <button>restore</button>
+            </td>
+          </tr>
+        </tbody>
+      </table>
     </div>
     <div v-if="isLoggedIn && !isVerified">
-      <router-link to="/verify">Verify Me!</router-link>
-    </div>
-    <div v-if="!isLoggedIn">
-      <router-link to="/about">Back</router-link>
+      <!-- TODO: ikoder - Add other features for this page -->
+      <router-link to="/verify"><button class="button is-primary">Verify Me!</button></router-link>
     </div>
-    <br />
-    <a class="button is-danger" v-if="isLoggedIn" @click="logout">Logout</a>
   </div>
 </template>
 
+<script src="https://cdn.jsdelivr.net/npm/sweetalert2@9"></script>
 <script>
+import shopOwnerNavbar from "../components/NavBar-ShopOwner";
+import axios from "axios";
+import Swal from "sweetalert2";
 import firebase from "firebase";
-
 export default {
-  name: "shopOwnerHome",
-
+  name: "MyShops",
+  components: {
+    shopOwnerNavbar
+  },
   data() {
     return {
-      name: "",
-      uid: "",
+      shopList: [],
+      searchResult: "",
+      renderComponent: true,
       isLoggedIn: false,
-      isVerified: false,
+      isVerified: false
     };
   },
+  methods: {
+    deleteshop: function(id) {
+      Swal.fire({
+        title: "Are you sure?",
+        text: "You won't be able to revert this!",
```

Comment:
If we have an option of restoring the shop upon deletion, why do we call out this message?

Reply:
Removed it!